### PR TITLE
Spam: use thresholds to decide actions

### PR DESCRIPTION
**Pull request by humitos**
- Create an admin filter to filter projects by SPAM threashold
- Return 401 static response when project passed the threshold
- Return a custom robots.txt with everything denied if passed the threshold

Example of a page returned if the project is detected as SPAM:

Screenshot_2021-10-28_18-28-16

Related to #5227

### CODE REVIEW

File: readthedocs/proxito/views/mixins.py
Comment (humitos) [reply readthedocs/proxito/views/mixins.py:183]: What about prefixing all these functions with `deny_`? eg. `deny_serving_docs`, `deny_showing_dashboard`, `deny_showing_ads`

Comment (humitos) [reply readthedocs/proxito/views/mixins.py:183]: or prefix with `dont_` as I did in the admin filters?

File: readthedocs/projects/admin.py
Comment (humitos) [inline readthedocs/projects/admin.py:152]:
```suggestion
    title = 'Spam Threshold'
```

File: readthedocs/templates/spam.html
Comment (humitos) [reply readthedocs/templates/spam.html]: Done! I had to bring the logo to readthedocs.org repository, tho.

File: readthedocs/templates/spam.html
Comment (humitos) [reply readthedocs/templates/spam.html]: I updated this to link to the support URL. However, we can't use `{% url %}` here because the URL is not defined in El Proxito.

**Comment by humitos** [comment]
@ericholscher 

> I don't think we're currently blocking dashboard access with this PR, but that's definitely not required

Yeah, I didn't do this yet because I'm not sure what we want to block. Should we only block `ProjectDetailView` from non-maintainer users? Or there are other logic we want to cover besides that?

File: readthedocs/proxito/views/mixins.py
Comment (agjohnson) [reply readthedocs/proxito/views/mixins.py:182]: @humitos This might be good to clear up before merge too.

I don't have opinions here, but will throw one more general suggestion in: using auxiliary verbs in the member naming.

`deny_serve_docs` - "deny" here is ambiguous, and could be pasred as a question ("is serve docs denied?") or a statement ("block this request from serving docs").

Adding auxiliary verbs clarifies this. `is_serve_docs_denied` parses more immediately as a function that checks if serve docs is allowed. Also neat: it implies a boolean operation and suggests a yes/no answer in return.

It's wordier, but it's clear.

It might save the next person time or confusion while scanning through the code as well. All these suggestions solve the overlap with the underlying `serve_docs` though, so any sound fine.

Comment (humitos) [reply readthedocs/proxito/views/mixins.py:182]: @agjohnson sold! I'll adapt this code to use that naming.

**Comment by ericholscher** [comment]
> Yeah, I didn't do this yet because I'm not sure what we want to block. Should we only block ProjectDetailView from non-maintainer users? Or there are other logic we want to cover besides that?

Seems reasonable. But let's not worry about it currently, it's a small detail we can work out if we need to. Blocking the docs pages is the most important part.

**Review by ericholscher — APPROVED** [review]

**Pull request merged**
